healthcare/api-client/v1/hl7v2/hl7v2_messages_test.py

```diff
@@ -0,0 +1,191 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+import sys
+import uuid
+
+from gcp_devrel.testing import eventually_consistent
+
+# Add datasets for bootstrapping datasets for testing
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'datasets')) # noqa
+import datasets
+import hl7v2_stores
+import hl7v2_messages
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+dataset_id = 'test_dataset_{}'.format(uuid.uuid4())
+hl7v2_store_id = 'test_hl7v2_store-{}'.format(uuid.uuid4())
+hl7v2_message_file = 'resources/hl7-sample-ingest.json'
+label_key = 'PROCESSED'
+label_value = 'TRUE'
+
+
+@pytest.fixture(scope='module')
+def test_dataset():
+    dataset = datasets.create_dataset(
+        project_id,
+        cloud_region,
+        dataset_id)
+
+    yield dataset
+
+    # Clean up
+    datasets.delete_dataset(
+        project_id,
+        cloud_region,
+        dataset_id)
+
+
+@pytest.fixture(scope='module')
+def test_hl7v2_store():
+    hl7v2_store = hl7v2_stores.create_hl7v2_store(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+    yield hl7v2_store
+
+    hl7v2_stores.delete_hl7v2_store(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+
+def test_CRUD_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.create_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.get_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    hl7v2_messages.delete_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    out, _ = capsys.readouterr()
+
+    # Check that create/get/list/delete worked
+    assert 'Created HL7v2 message' in out
+    assert 'Name' in out
+    assert 'Deleted HL7v2 message' in out
+
+
+def test_ingest_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
```

Comment:
I thought this test used to be eventually consistent. Has the situation changed?

Reply:
I thought the list_messages part was made eventually consistent because you thought the list_messages method was flaky, but it wasn't flaky--it was broken for a short amount of time because the response body from the method changed.

I don't see much reason to keep it eventually consistent but please let me know if you do.

Reply:
Sounds good :)

FYI, to verify whether the test is flaky or not, you can utilize flaky. You can add a decorator like the following temporarily:

```python
@pytest.mark.flaky(max_runs=100, min_passes=100)
```

Then you can verify this locally:
```bash
$ python -m nox -s py-3.6 -- hl7v2_messages_test.py
```